src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs

```diff
@@ -83,7 +83,7 @@ internal string GetDefaultStorageConnectionString()
         // This method should not be called before the app settings are resolved into the options.
         // Because of this, we wait to validate the options until right before building a durability provider, rather
         // than in the Factory constructor.
-        private void EnsureInitialized()
+        private void EnsureInitialized(bool isExternalClient = false)
```

Comment:
What are the values of this.options and this.azureStorageOptions when we are using external clients? I'm just wondering if we even need to try and validate those for external client scenarios.

One other niche edge case about setting `this.hasValidationOptions=true` for external clients is the case where the customer has both external clients and a a default durability provider used in the same app. In that case, if the external client is created first (which may not be possible?), we would never instantiate this.defaultSettings. This wouldn't likely have a huge impact, but we may just be better renaming this method `EnsureDefaultClientSettingsInitialized()` and then only call it if ExternalClient is false.

Reply:
They have default values. For example, `this.options.HubName` is "TestTaskHub" and `this.azureStorageOptions.ConnectionStringName` is `null`. `EnsureInitialized()` doesn't seem to have an impact on external clients because there's already a check for a task hub name in `CreateClient()` from `DurableClientFactory`. The storage connection string name check happens in `GetAzureStorageOrchestrationServiceSettings()` though.

I see, I'll rename the method and add a check in `GetDurabilityProvider(attribute)` to only call `EnsureDefaultClientSettingsInitialized()` if it's not an external client.